app/controllers/graders_controller.rb

```diff
@@ -253,10 +253,10 @@ def unassign_graders(grouping_ids, grader_ids)
 
   # Returns array of grouping ids with non empty submissions
   def filter_empty_submissions(grouping_ids)
-    grouping_ids.select do |grouping_id|
-      submission = Submission.find_by(grouping_id: grouping_id, submission_version_used: true)
-      submission && !submission.is_empty
-    end
+    submissions = Submission.where(grouping_id: grouping_ids, submission_version_used: true)
+                            .pluck(:grouping_id, :is_empty)
+    non_empty_grouping_ids = submissions.reject { |_, is_empty| is_empty }.map { |grouping_id, _| grouping_id.to_s }
+    non_empty_grouping_ids & grouping_ids
```

Comment:
The `& grouping_ids` isn't necessary, the `where` filter should only include grouping ids that were included in the original `grouping_ids` argument